src/Recommendations/Process/Hydration.php

```diff
@@ -14,64 +14,80 @@
+use eLife\Bus\Queue\SingleItemRepository;
 use eLife\Recommendations\RuleModel;
-use function GuzzleHttp\Promise\all;
 
 final class Hydration
 {
-    use GetSdk;
-
-    /** @var ApiSdk */
-    private $sdk;
     private $cache = [];
+    private $repo;
+    private $sdk;
 
-    public function transform(RuleModel $item)
+    public function __construct(ApiSdk $sdk, SingleItemRepository $repo)
     {
-        $sdk = $this->getSdk($item->getType());
-        $entity = $sdk->get($item->getId());
-
-        return $entity->wait(true);
+        $this->repo = $repo;
+        $this->sdk = $sdk;
     }
 
-    public function __construct(ApiSdk $sdk)
+    public function convertType(string $type): string
     {
-        $this->sdk = $sdk;
+        switch ($type) {
+            case 'article':
+            case 'correction':
+            case 'editorial':
+            case 'feature':
+            case 'insight':
+            case 'research-advance':
+            case 'research-article':
+            case 'research-exchange':
+            case 'retraction':
+            case 'registered-report':
+            case 'replication-study':
+            case 'short-report':
+            case 'tools-resources':
+                return 'article';
+            default:
+                return $type;
+        }
     }
 
-    /**
-     * This is quite a basic model conversion, if we start offering
-     * recommendations for other items such as podcast episodes then
-     * this will have to change. (getId doesn't exist on podcasts).
-     */
-    private function getModel(RuleModel $model, $unwrap = false)
+    public function getPodcastEpisodeChapterById($id): PodcastEpisodeChapterModel
     {
-        if (isset($this->cache[$model->getType()][$model->getId()])) {
-            return $this->cache[$model->getType()][$model->getId()];
-        }
-        $sdk = $this->getSdk($model->getType());
-        $entity = $sdk->get($model->getId());
-        if ($unwrap) {
-            return $entity->wait(true);
-        }
+        list($episodeId, $chapterId) = explode('-', $id);
+        // I want to be able to do this in list :(
+        $episodeId = (int) $episodeId;
+        $chapterId = (int) $chapterId;
+
+        /** @var PodcastEpisode $episode */
+        $episode = $this->repo->get('podcast-episode', $episodeId);
+        $chapter = $episode
+                ->getChapters()
+                ->filter(function (PodcastEpisodeChapter $chapter) use ($chapterId) {
+                    return $chapter->getNumber() === $chapterId;
+                })
+                ->toArray()[0] ?? null;
 
-        return $entity->then(function (ArticleVersion $model) {
-            $this->cache[$model->getType()][$model->getId()] = $model;
-            // @todo enable if required.
-            // $this->extractRelatedFrom(new RuleModel($model->getId(), $model->getType()));
-            return $model;
-        });
+        return new PodcastEpisodeChapterModel($episode, $chapter);
```

Comment:
@thewilkybarkid This is my current implementation of `getPodcastEpisodeChapterById `

There might be a better way though